tests/testthat/test-input-csv-functions.R

```diff
@@ -298,3 +298,17 @@ test_that("read/scan: arg 'decimal_comma' works", {
     pl$DataFrame(a = c(1.5, 2), b = c("a", NA), c = c(2L, NA))$cast(c = pl$Int64)
   )
 })
+
+test_that("can read compressed CSV files", {
+  skip_if_not_installed("R.utils")
+
+  df <- pl$DataFrame(col1 = letters, col2 = 1:26)
+  path <- withr::local_tempfile(fileext = "csv")
+  df$write_csv(path)
+
+  gz_path <- R.utils::gzip(path)
```

Comment:
Would this be a temp file?

Aside from the possibility that this will not be removed, I believe that compressed CSV files can be generated using the data.table package, so no additional dependencies are needed.
https://rdatatable.gitlab.io/data.table/reference/fwrite.html